detection_rules/rule.py

```diff
@@ -331,6 +333,19 @@ def validate_note(self):
         if self.setup_in_note and self.setup:
             raise ValidationError("Setup header found in both note and setup fields.")
 
+    def validate_new_terms_fields(self, meta: RuleMeta) -> None:
```

Comment:
> Is the history_window_start an appropriate range

For `history_window_start` I will have to dig into validation on this more, maybe from Kibana. At the moment I did not locate any validation for the `from` field either than it's definition which accepts the same values in Kibana.

> Are the fields provided only when "type": "new_terms" (this could also be a unit test)

I don't believe a unit test is necessary here. If `new_terms` is not a valid rule type, it will fail the Marshmallow dataclass checks for `TOMLRuleContents`. If it is `new_terms`, the schema should raise any errors with the fields or values, including the names of the TOML nodes where these values are defined. Also, on `rule.post_validation`, I added a [line](https://github.com/elastic/detection-rules/blob/2ff13231ae5a087a8799ded4b28a469730225e01/detection_rules/rule.py#L937) to check for a validation method and call it if it exists. The `validation` method inside of the `NewTermsRuleData` class is then called which does further validation on the rule itself.